client/nginx.go

```diff
@@ -469,32 +470,54 @@ func (client *NginxClient) DeleteHTTPServer(upstream string, server string) erro
 // UpdateHTTPServers updates the servers of the upstream.
 // Servers that are in the slice, but don't exist in NGINX will be added to NGINX.
 // Servers that aren't in the slice, but exist in NGINX, will be removed from NGINX.
-func (client *NginxClient) UpdateHTTPServers(upstream string, servers []UpstreamServer) ([]UpstreamServer, []UpstreamServer, error) {
+// Servers that are in the slice and exist NGINX, but have different parameters, will be updated.
+func (client *NginxClient) UpdateHTTPServers(upstream string, servers []UpstreamServer) (added []UpstreamServer, deleted []UpstreamServer, updated []UpstreamServer, err error) {
```

Comment:
I was thinking (cc @pleshakov, @Dean-Coakley) The way we  know if a given server is already in NGINX is by the `UpstreamServer.Server` which is the address of the server.

However, if we configure this upstream:
```
upstream test {
    server 127.0.0.2;
}
``` 

And we make a call to the API:
```
$ curl http://nginxplus/api/4/http/upstreams/test/servers/0
{"id":0,"server":"127.0.0.2:80","weight":1,"max_conns":0,"max_fails":1,"fail_timeout":"10s","slow_start":"0s","route":"","backup":false,"down":false}
```
The NGINX Plus API returns the `ip:port` 

This means, if we want to do an update like
`client.UpdateHTTPServers([]UpstreamServer{ Server: "127.0.0.2"})` The method will tell us that it wasn't updated, but 1 server was deleted and another one was added. The reason is that `"127.0.0.2" != "127.0.0.2:80"` (note the port)

I was wondering, should we return an error when the upstream.Server has no port but just IP? or is it ok to have this little inconsistency? 

Another way would be to assume `:80` if no port is selected in upstream.Server

Thoughts?

Reply:
@Rulox good catch.

I like ` we return an error when the upstream.Server has no port but just IP? `